sql/core/src/test/scala/org/apache/spark/sql/DatasetPrimitiveSuite.scala

```diff
@@ -28,6 +28,8 @@ case class SeqClass(s: Seq[Int])
 
 case class ListClass(l: List[Int])
 
+case class JListClass(l: java.util.List[Int])
```

Comment:
it's planned but no one is working on it yet. You can start by creating a JIRA ticket if you are interested :)